dynawo/sources/Models/Modelica/Dynawo/Electrical/StaticVarCompensators/SVarCPV.mo

```diff
@@ -0,0 +1,80 @@
+  parameter Types.PerUnit BShuntPu "Fixed susceptance of the static var compensator in p.u (for standby mode) (base SnRef)";
+  parameter Types.VoltageModule UNom "Static var compensator nominal voltage in kV";
+  final parameter Types.VoltageModule UThresholdUpPu =  UThresholdUp / UNom;
+  final parameter Types.VoltageModule UThresholdDownPu =  UThresholdDown / UNom;
+
+  Modelica.Blocks.Interfaces.RealInput URef(start = URef0) "Voltage reference for the regulation in kV";
+  Modelica.Blocks.Interfaces.BooleanInput selectModeAuto(start = selectModeAuto0) "Whether the static var compensator is in automatic configuration";
+  Modelica.Blocks.Interfaces.IntegerInput setModeManual(start = setModeManual0) "Mode selected when in manual configuration";
+
+  Types.PerUnit BVarRawPu(start = BVar0Pu) "Raw variable susceptance of the static var compensator in p.u (base SnRef)";
+  Types.PerUnit BVarPu(start = BVar0Pu) "Variable susceptance of the static var compensator in p.u (base SnRef)";
+  Types.PerUnit BPu(start = B0Pu) "Susceptance of the static var compensator in p.u (base SnRef)";
+  Types.VoltageModulePu UPu(start = U0Pu) "Voltage amplitude at terminal in p.u (base UNom)";
+  Types.ReactivePowerPu QInjPu(start = B0Pu * U0Pu ^ 2) "Reactive power in p.u (base SnRef) (generator convention)";
+
+  BaseControls.ModeHandling modeHandling(Mode0 = Mode0, UNom = UNom, URefDown = URefDown, URefUp = URefUp, UThresholdDown = UThresholdDown, UThresholdUp = UThresholdUp, tThresholdDown = tThresholdDown, tThresholdUp = tThresholdUp, URef0 = URef0);
+
+protected
+  parameter Types.PerUnit B0Pu "Start value of the susceptance in p.u (base SnRef)";
+  parameter Types.VoltageModulePu U0Pu  "Start value of voltage amplitude at injector terminal in p.u (base UNom)";
+  parameter Types.ComplexVoltagePu u0Pu  "Start value of complex voltage at injector terminal in p.u (base UNom)";
+  parameter Types.ComplexCurrentPu i0Pu  "Start value of complex current at injector terminal in p.u (base UNom, SnRef) (receptor convention)";
+  parameter Types.VoltageModule URef0  "Start value of voltage reference in kV";
+  final parameter Types.PerUnit BVar0Pu = B0Pu - BShuntPu "Start value of variable susceptance in p.u";
+  parameter BaseControls.Mode Mode0 "Start value for mode";
+  parameter Boolean selectModeAuto0 = true "Start value of the boolean indicating whether the SVarC is initially in automatic configuration";
+  parameter Integer setModeManual0 = 2 "Start value of the mode when in manual configuration";
+
+equation
+  UPu = Modelica.ComplexMath.'abs'(terminal.V);
+
+  UPu = modeHandling.UPu;
+  URef = modeHandling.URef;
+  selectModeAuto = modeHandling.selectModeAuto;
+  setModeManual = modeHandling.setModeManual;
+
+  modeHandling.URefPu = UPu + Lambda * (SystemBase.SnRef / SNom) * (BVarRawPu + BShuntPu) * UPu ^ 2;
+  BVarPu = if BVarRawPu > BMaxPu - BShuntPu then BMaxPu - BShuntPu elseif BVarRawPu < BMinPu - BShuntPu then BMinPu - BShuntPu else BVarRawPu;
```

Comment:
Again this question of variable vs total susceptance.

Reply:
Corrected, thanks !